sigal/plugins/encrypt/encrypt.py

```diff
@@ -0,0 +1,297 @@
+                    except Exception as e:
+                        logger.error("Encryption failed for %s: %s", f, e)
+                    else:
+                        logger.info("Encrypting %s...", f)
+                        try:
+                            with open(full_path, "wb") as outfile:
+                                outfile.write(outBuffer.getbuffer())
+                            cacheEntry.add(f)
+                        except Exception as e:
+                            logger.error("Could not write to file %s: %s", f, e)
+
+def fix_html(gallery, settings, config, albums):
+    logger = gallery.logger
+    if gallery.settings["write_html"]:
+        decryptorConfigTemplate = """
+        Decryptor.init({{
+            password: "{filtered_password}",
+            worker_script: "{worker_script}",
+            galleryId: "{galleryId}",
+            gcm_tag: "{escaped_gcm_tag}", 
+            kdf_salt: "{escaped_kdf_salt}", 
+            kdf_iters: {kdf_iters} 
+        }});
+        """
+        config["filtered_password"] = "" if config.get("ask_password", False) else config["escaped_password"]
+
+        with progressbar(albums.values(), label="%16s" % "Fixing html files", file=gallery.progressbar_target, show_eta=True) as albums:
+            for album in albums:
+                index_file = os.path.join(album.dst_path, album.output_file)
+                contents = None
+                with open(index_file, "r", encoding="utf-8") as f:
+                    contents = f.read()
+                root = BeautifulSoup(contents, "html.parser")
+                head = root.find("head")
+                if head.find(id="_decrypt_script"):
+                    head.find(id="_decrypt_script").decompose()
+                if head.find(id="_decrypt_script_config"):
+                    head.find(id="_decrypt_script_config").decompose()
+                theme_path = os.path.join(settings["destination"], 'static')
+                theme_url = url_from_path(os.path.relpath(theme_path, album.dst_path))
+                scriptNode = root.new_tag("script", id="_decrypt_script", src="{url}/decrypt.js".format(url=theme_url))
+                scriptConfig = root.new_tag("script", id="_decrypt_script_config")
+                config["worker_script"] = "{url}/decrypt-worker.js".format(url=theme_url)
+                decryptorConfig = decryptorConfigTemplate.format(**config)
+                scriptConfig.append(root.new_string(decryptorConfig))
+                head.append(scriptNode)
+                head.append(scriptConfig)
+                with open(index_file, "w", encoding="utf-8") as f:
+                    f.write(root.prettify())
+
+def copy_assets(settings):
+    theme_path = os.path.join(settings["destination"], 'static')
+    for root, dirs, files in os.walk(ASSETS_PATH):
+        for file in files:
+            copy(os.path.join(root, file), theme_path, symlink=False, rellink=False)
```

Comment:
You have only two files to copy, so maybe use their path directly here? Would be more explicit.

Reply:
Sure.

Reply:
Changed in 181d0bb